framework/src/node/consensus/consensus.ts

```diff
@@ -451,7 +454,14 @@ export class Consensus {
 			finalizedHeight = bftVotes.maxHeightPrecommitted;
 		}
 
-		await this._chain.saveBlock(block, stateStore, finalizedHeight, {
+		// Result Validation
+		// Verify validatorsHash
+		await this._verifyValidatorsHash(apiContext, block);
+		// Verify stateRoot
+		const currentState = await this._prepareStore(stateStore);
```

Comment:
I cannot  think of nicer name, but `currentState`  and `_prepareStore` still doesnt match. maybe `_prepareFinallizingState` would be bit better?

Reply:
yeah I think `_prepareFinalizingState` store sounds better